scripting/2.0/modules/ROOT/pages/edits/scripting-module-upgrade-migrate.adoc

```diff
@@ -0,0 +1,109 @@
+= Upgrade Scripting Module to Version 2.0 - Mule 4
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Upgrade Scripting Module to version 2.0.0.
+
+== Supported Upgrade Paths
+
+[%header,cols="50a,50a"]
+|===
+|From Version | To Version
+|1.1.7 |2.0.0
+|===
+
+== Changes in This Release
+
+This release contains the following changes:
+
+* Scripting Module no longer provides default scripting engines:
+** Use the *Configure...* helper button to install your desired engines.
+** Specify the desired engine in the `pom.xml` file of your application.
+* For any data stream provided in the payload or variables, a cursor object opens and binds, preventing the need for you to open new cursors. Any `openCursor()` provided to the Scripting Module closes after the script execution.
+
+
+== Upgrade Prerequisites
+
+Before you perform the upgrade, you must complete the following:
+
+* Create a backup of your files, data, and configurations in case you need to restore to them.
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade:
+
+. In Anypoint Studio, create a Mule project.
+. In *Mule Palette*, click *Search in Exchange*.
+. In *Add Dependencies to Project*, enter  in the search field.
+. In *Available modules*, select `Scripting Module` and click *Add*.
+. Click *Finish*.
+. Verify that the `mule-scripting-module` dependency version is `2.0.0` in the file in the Mule project.
+
+Studio upgrades the connector version automatically.
+
+
+== Post Upgrade Steps
+
+After you install the latest version of the connector, add a `jsr-223` compliant scripting engine:
+
+. In Anypoint Studio, navigate to the *Execute* tab operation of your application.
+. Use one of the following two options to install and configure a `jsr-223` compliant engine: +
+
+* In the *Required Libraries* section, click *Configure...* and select one of the following options to search and install the engine: +
+
+** Add recommended libraries +
+Install Groovy engine as recommended. This option only appears if you did not previously install Groovy.
+** Use local file +
+Browse a local file for the required engine library and install it.
+** Add Maven dependency +
+Browse the engine dependency and install it. +
+
+* In the `pom.xml` file, manually add the engine in both of these sections:
+** `<sharedLibraries>` tags under the `<configuration>` section
+** `<dependency>` tags under the `<dependencies>` section
+
+[start=5]
+. In the *General* section of the *Execute* tab, click the refresh button to update the engine values list for the operation.
+. In *Engine*, select the installed engine.
+
+
+=== Verify the Upgrade
+
+After you install the latest version of the connector, verify the upgrade:
```

Comment:
Does the user perform this before or after post-upgrade steps? Clarify.
Also, I would make this a separate, peer task, or add it to the Upgrade task as the final step with substeps, but I would not make it a solitary subtopic.